### ISSUE TITLE
after updating to latest podenv fails with: Invalid default application: Error: Invalid Dhall.Decoder

### ISSUE THREAD

**Issue by juhp** [podenv/podenv#40]
I guess this is expected due to the API change in 7a28378f:
```ShellSession
$ podenv image:ubi8
podenv: Invalid default application: Error: Invalid Dhall.Decoder                                               
                                                                                
Every Decoder must provide an extract function that does not fail with a type   
error if an expression matches the expected type.  You provided a Decoder that  
disobeys this contract                                                          
                                                                                
The Decoder provided has the expected dhall type:                               
                                                                                
↳ { host_network : Bool, hostfile : Bool }
                                                                                
and it threw a type error during extraction from the well-typed expression:     
                                                                                
↳ { alsa = False
  , cwd = False
  , dbus = False
  , dri = False
  , gitroot = False
  , gpg = False
  , hostfile = False
  , interactive = False
  , keep = False
  , kerberos = False
  , kvm = False
  , network = False
  , pipewire = False
  , pulseaudio = False
  , root = False
  , ssh = False
  , terminal = False
  , tun = False
  , video = False
  , wayland = False
  , x11 = False
  }
                                                                                

CallStack (from HasCallStack):
  error, called at src/Relude/Debug.hs:218:11 in relude-0.7.0.0-FtIHw33t24H7i0K6Sc8qVf:Relude.Debug
  error, called at src/Podenv/Config.hs:241:16 in podenv-0.1.0.0-8uaivfntuKMDUYHGXQZDiT:Podenv.Config
```
but I am just wondering if something could be done to make this smoother.
With my little dhall knowledge/experience I see a couple of possible options
how to update to podenv/hub@00c3dfe90273b2eb09924b1feca7da81b7d478ce in this case:

In a way it would be good to bundle Hub.dhall with podenv (perhaps as a data-file)
though this also would make podenv a bit harder to deploy (embedding the file inside
the binary would be even cleaner). Alternatively maybe podenv could detect that
the Hub.dhall hash is out of date and auto-update it say.
I dunno if dhall has some update mechanisms too...?
Another workaround might be to have an `update` or `sync` option/command for this?

**Comment by juhp** [comment]
Actually it looks like something needs to be updated, since I getting the API mismatch error now even after updating or initialising?

**Comment by TristanCacqueray** [comment]
It seems like you had two issues.

The main one should be addressed by #39 . As you suggested, this PR effectively bundles the hub with podenv. I think this is the safest thing to do for now, and we can revisit later what should be included by default.

The other issue is that the `Podenv.Dhall` uses template haskell to generate Haskell types from the Dhall schemas, and that does not rebuild when the schemas change. I don't know how to fix that, and in the meantime a clean rebuild (or updating a comment in `Podenv.Dhall`) mitigate the issue.
